app/src/common/shared/com/igalia/wolvic/ui/widgets/Windows.java

```diff
@@ -1360,6 +1365,14 @@ private void openNewTab(@NonNull String aUri, @NewTabLocation int aLocation) {
         }
     }
 
+    public void openInKioskMode(@NonNull String aUri) {
+        Session session = SessionStore.get().createSuspendedSession(aUri, false);
+        setFirstPaint(mFocusedWindow, session);
+        mFocusedWindow.setSession(session, WindowWidget.DEACTIVATE_CURRENT_SESSION);
+        mFocusedWindow.setKioskMode(true);
+        mFocusedWindow.setIsFullScreen(true);
```

Comment:
I guess setting the window to fullscreen is something that it's debatable.

Reply:
Sure, we can change that. This was just a convenient way to make sure that other windows are hidden, etc.